src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Constructor.cs

```diff
@@ -333,7 +361,218 @@ static void Main()
             }
             """;
 
-        CompileAndVerify(source, expectedOutput: IncludeExpectedOutput("default-4,default-10,custom-4,both-20"));
+        var verifier = CompileAndVerify(source, expectedOutput: IncludeExpectedOutput("default-4,default-10,custom-4,both-20"));
+        var compilation = (CSharpCompilation)verifier.Compilation;
+
+        var semanticModel = compilation.GetSemanticModel(compilation.SyntaxTrees.Single());
+        var root = semanticModel.SyntaxTree.GetRoot();
+
+        var withElements = root.DescendantNodes().OfType<WithElementSyntax>().ToArray();
```

Comment:
nit: this is another instance where I would have used the `GetSyntax` helper: `GetSyntax<WithElementSyntax>(tree, "with()")` gives an unambiguous syntax node and there is no risk of messing with indexes